### PR TITLE
Apache webserver handles TLS on Ironic

### PR DESCRIPTION
**Pull request by namnx228**
This PR let Apache web server handle TLS on Ironic
Related PRs:
metal3-io/ironic-image#230 (This PR needs to be merged before the current PR can pass the CI)
metal3-io/ironic-inspector-image#70

### CODE REVIEW

**Comment by dhellmann** [comment]
Do we want to use a "reverse proxy" or mod_wsgi? @dtantsur what is the recommendation for deploying Ironic outside of kubernetes?

If we don't want to use mod_wsgi, is there any reason not to use a sidecar for the TLS proxy?

**Comment by namnx228** [comment]
> Do we want to use a "reverse proxy" or mod_wsgi? @dtantsur what is the recommendation for deploying Ironic outside of kubernetes?
> 
> If we don't want to use mod_wsgi, is there any reason not to use a sidecar for the TLS proxy?

@dhellmann Yes, we use mod_wsgi for Ironic-api, and use proxy for Ironic-conductor and ironic inspector. The proxy is deployed in the same container with the application (the conductor, the inspector) to improve security.

File: ironic-deployment/ironic/ironic.yaml [hunk 1]
```diff
@@ -104,6 +90,14 @@ spec:
           envFrom:
             - configMapRef:
                 name: ironic-bmo-configmap
+        - name: httpd-reverse-proxy
+          image: quay.io/metal3-io/ironic-inspector
```
Comment (kashifest) [inline ironic-deployment/ironic/ironic.yaml:94]: Should it be inspector image or ironic image?

Comment (kashifest) [reply ironic-deployment/ironic/ironic.yaml:94]: I think I got the answer. It was added here https://github.com/metal3-io/ironic-inspector-image/commit/7e2757d890917dc7256bbae1b23fed00c8e74b73#diff-9f9f6a924db88783257f0da0c2ae75be9d01b5b1ea71d522caaafab42362392f

File: ironic-deployment/ironic/ironic.yaml [hunk 2]
```diff
@@ -42,20 +42,6 @@ spec:
...
-             capabilities:
-               add: ["NET_ADMIN"]
-          command:
-            - /bin/runhttpd
-          volumeMounts:
-            - mountPath: /shared
-              name: ironic-data-volume
-          envFrom:
-            - configMapRef:
-                name: ironic-bmo-configmap
```
Comment (maelk) [inline ironic-deployment/ironic/ironic.yaml:58]: you should not remove this. This is the webserver for IPA image and other PXE related files

**Review by maelk — CHANGES_REQUESTED** [review]

File: ironic-deployment/ironic/ironic.yaml (hunk 2 above)
Comment (namnx228) [reply ironic-deployment/ironic/ironic.yaml:58]: @maelk I remove it because the httpd server which works as a WSGI server inside the ironic-api pod will also serve the IPA image.

File: tools/deploy.sh [hunk 3]
```diff
@@ -13,6 +13,7 @@ if [ -z "${1:-}" ] || [ -z "${2:-}" ] || [ -z "${3:-}" ] || [ -z "${4:-}" ] || [
     exit 1
 fi
 
+
```
Comment (elfosardo) [inline tools/deploy.sh:16]: nit: unrelated change

File: tools/run_local_ironic.sh [hunk 4]
```diff
@@ -17,9 +17,11 @@ PROVISIONING_INTERFACE="${PROVISIONING_INTERFACE:-"ironicendpoint"}"
 CLUSTER_DHCP_RANGE="${CLUSTER_DHCP_RANGE:-"172.22.0.10,172.22.0.100"}"
 IRONIC_KERNEL_PARAMS="${IRONIC_KERNEL_PARAMS:-"console=ttyS0"}"
 
+
```
Comment (elfosardo) [inline tools/run_local_ironic.sh:20]: nit: ditto

**Review by elfosardo — APPROVED** [review]

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

**Review by maelk — APPROVED** [review]

**Comment by namnx228** [comment]
Thanks very much everyone for your review and approval.
